core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -240,15 +244,25 @@ class UnitAutomation{
 
     private fun tryAdvanceTowardsCloseEnemy(unit: MapUnit): Boolean {
         // this can be sped up if we check each layer separately
-        var closeEnemies = unit.getTile().getTilesInDistance(5)
-                .filter{ containsAttackableEnemy(it, MapUnitCombatant(unit)) && unit.movement.canReach(it)}
+        var closeEnemies = getAttackableEnemies(
+            unit,
+            unit.movement.getDistanceToTilesWithinTurn(
+                unit.getTile().position,
+                unit.getMaxMovement() * 3f
+            ),
+            tiles = unit.getTile().getTilesInDistance(5)
+        ).filter {
```

Comment:
We'll probably get a lot of duplicates here, because we have (for each unit in range) several places where we can stand in order to attack him, so we should probably do .map{it.tileToAttack}.distinct() and that way we'll calculate each tile once

Reply:
Doesn't tile you attack from affect the damage? So there might be case when attacking from one tile kills you, but from another does not? I guess it's not true for barbarians, but is true for AI civs.

Reply:
Not really, the tile that you're attacking makes a big difference but not where you're attacking from iirc

Reply:
But I'm not 100% sure, it doesn't look like that big of a performance hit so it's probably fine

Reply:
General adjastency bonus, for example